src/Service/Resolvers/CosmosMutationEngine.cs

```diff
@@ -57,6 +64,11 @@ private async Task<JObject> ExecuteAsync(IDictionary<string, object?> queryArgs,
             Container container = client.GetDatabase(resolver.DatabaseName)
                                         .GetContainer(resolver.ContainerName);
 
+            // If authorization fails, an exception will be thrown and request execution halts.
+            string graphQLType = context.Selection.Field.Type.NamedType().Name.Value;
+            string entityName = _metadataProvider.GetEntityName(graphQLType);
+            AuthorizeMutationFields(context, queryArgs, entityName, resolver.OperationType);
```

Comment:
For SQL, as far as I remember, we add the @authorize directive for the fields in the mutation. 

Why can't we follow similar approach here? 

 I may be remembering it wrong - can you please clarify

Reply:
nvm I found AuthorizeMutationFields for SqlMutationEngine too. Looks like `authorize` directive is used only for Query ie. read

Reply:
> "Invalid filter object nested field input value type.",

The authorize directive, e.g. `@authorize(roles: [ "authenticated", "myRole" ])`, would need to be included in the provided GraphQL schema for object type / object field read permissions

Reply:
Your follow up comment is correct that the authorize directive is used for read:
https://chillicream.com/docs/hotchocolate/v12/security/authorization#roles

Reply:
Yes, for Cosmos, the ```@authorize``` directive needs to be included in the user provided ```schema.gql``` file to use it on Query and Query fields.